```
/ClickHouse/base/poco/MongoDB/src/Connection.cpp
2023-02-27 17:33:54 In file included from ../../ClickHouse/base/poco/MongoDB/src/Connection.cpp:16:
2023-02-27 17:33:54 /ClickHouse/base/poco/MongoDB/include/Poco/MongoDB/Connection.h:165:36: error: return type of out-of-line definition of 'Poco::MongoDB::Connection::uri' differs from that in the declaration
2023-02-27 17:33:54     inline std::string Connection::uri() const
2023-02-27 17:33:54            ~~~~~~~~~~~             ^
2023-02-27 17:33:54 /ClickHouse/base/poco/MongoDB/include/Poco/MongoDB/Connection.h:94:29: note: previous declaration is here
2023-02-27 17:33:54         const std::string & uri() const;
2023-02-27 17:33:54       
```

--- a/base/poco/MongoDB/include/Poco/MongoDB/Connection.h
+++ b/base/poco/MongoDB/include/Poco/MongoDB/Connection.h
@@ -162,7 +162,7 @@ namespace MongoDB
     {
         return _address;
     }
-    inline std::string Connection::uri() const
+    inline const std::string & Connection::uri() const
     {
     	return _uri;
     }

--- a/src/Dictionaries/MongoDBDictionarySource.cpp
+++ b/src/Dictionaries/MongoDBDictionarySource.cpp
@@ -118,7 +118,7 @@ MongoDBDictionarySource::MongoDBDictionarySource(
         Poco::MongoDB::Connection::SocketFactory socket_factory;
         connection->connect(uri, socket_factory);
 
-        Poco::URI poco_uri(connection.uri());
+        Poco::URI poco_uri(connection->uri());
 
         // Parse database from URI. This is required for correctness -- the
         // cursor is created using database name and collection name, so we have

